### PR TITLE
IDC: move unit tests from Jetpack to the IDC package.

### PR DESCRIPTION
**Pull request by kbrown9**
#### Changes proposed in this Pull Request:
* Move IDC tests from Jetpack to the identity-crisis package.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
Verify that the unit tests passed.

### CODE REVIEW

File: projects/packages/identity-crisis/tests/php/test-identity-crisis.php [hunk 1]
```diff
@@ -36,4 +54,264 @@ public function test_clear_all_idc_options_clears_expected() {
...
+		add_filter( 'jetpack_sync_idc_optin', '__return_false' );
+		$result = Identity_Crisis::sync_idc_optin();
+		remove_filter( 'jetpack_development_version', '__return_true' );
+		remove_filter( 'jetpack_sync_idc_optin', '__return_false' );
+
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * Test that that the jetpack_sync_idc_optin filter cast values to a bool.
```
Comment (kbrown9) [inline projects/packages/identity-crisis/tests/php/test-identity-crisis.php:83]:
```suggestion
	 * Test that the jetpack_sync_idc_optin filter cast values to a bool.
```

File: projects/packages/identity-crisis/tests/php/test-identity-crisis.php [hunk 2]
```diff
@@ -36,4 +54,264 @@ public function test_clear_all_idc_options_clears_expected() {
...
+	/**
+	 * Test that sync_idc_optin returns true when the JETPACK_SYNC_IDC_OPTIN constant is true.
+	 */
+	public function test_idc_optin_true_when_constant_true() {
+		Constants::set_constant( 'JETPACK_SYNC_IDC_OPTIN', true );
+		$this->assertTrue( Identity_Crisis::sync_idc_optin() );
+	}
+
+	/**
+	 * Test that sync_idc_optin resturns false when the JETPACK_SYNC_IDC_OPTIN constant is false.
```
Comment (kbrown9) [inline projects/packages/identity-crisis/tests/php/test-identity-crisis.php:102]:
```suggestion
	 * Test that sync_idc_optin returns false when the JETPACK_SYNC_IDC_OPTIN constant is false.
```

File: projects/packages/identity-crisis/tests/php/test-identity-crisis.php [hunk 3]
```diff
@@ -36,4 +54,264 @@ public function test_clear_all_idc_options_clears_expected() {
...
+	 * matches the expected value.
+	 */
+	public function test_sync_error_idc_validation_returns_true_when_option_matches_expected() {
+		add_filter( 'jetpack_sync_idc_optin', '__return_true' );
+		Jetpack_Options::update_option( 'sync_error_idc', Identity_Crisis::get_sync_error_idc_option() );
+
+		$result = Identity_Crisis::validate_sync_error_idc_option();
+
+		Jetpack_Options::delete_option( 'sync_error_idc' );
+
```
Comment (kbrown9) [inline projects/packages/identity-crisis/tests/php/test-identity-crisis.php:140]:
```suggestion
```

File: projects/packages/identity-crisis/tests/php/test-identity-crisis.php [hunk 4]
```diff
@@ -36,4 +54,263 @@ public function test_clear_all_idc_options_clears_expected() {
...
+		$this->assertTrue( $result );
+	}
+
+	/**
+	 * Test that get_sync_error_idc_option sanitizes out www and the protocol.
+	 */
+	public function test_get_sync_idc_option_sanitizes_out_www_and_protocol() {
+		// Don't need these filters for this test.
+		remove_filter( 'jetpack_sync_home_url', array( $this, 'return_example_url' ) );
+		remove_filter( 'jetpack_sync_site_url', array( $this, 'return_example_url' ) );
```
Comment (kbrown9) [inline projects/packages/identity-crisis/tests/php/test-identity-crisis.php:250]:
```suggestion
```

File: projects/packages/identity-crisis/tests/php/test-identity-crisis.php [hunk 5]
```diff
@@ -36,4 +54,260 @@ public function test_clear_all_idc_options_clears_expected() {
...
+		remove_filter( 'jetpack_sync_error_idc_validation', '__return_false' );
+
+		$this->assertTrue( $result );
+	}
+
+	/**
+	 * Test that get_sync_error_idc_option sanitizes out www and the protocol.
+	 */
+	public function test_get_sync_idc_option_sanitizes_out_www_and_protocol() {
+
```
Comment (kbrown9) [inline projects/packages/identity-crisis/tests/php/test-identity-crisis.php:248]:
```suggestion
```

File: projects/packages/identity-crisis/tests/php/test-identity-crisis.php [hunk 6]
```diff
@@ -36,4 +54,259 @@ public function test_clear_all_idc_options_clears_expected() {
...
+	}
+
+	/**
+	 * Test that get_sync_error_idc_option santizes out the protocol when there is an ip address
+	 * in the option.
+	 */
+	public function test_get_sync_idc_option_with_ip_address_in_option() {
+		remove_filter( 'jetpack_sync_home_url', array( $this, 'return_example_url' ) );
+		remove_filter( 'jetpack_sync_site_url', array( $this, 'return_example_url' ) );
+
```
Comment (kbrown9) [inline projects/packages/identity-crisis/tests/php/test-identity-crisis.php:275]:
```suggestion
```

**Review by samiff — APPROVED** [review]
Looks good and the tests are passing 👍

**Pull request merged**